agent/src/main/scala/za/co/absa/atum/agent/model/Measure.scala

```diff
@@ -17,182 +17,156 @@
 package za.co.absa.atum.agent.model
 
 import org.apache.spark.sql.functions._
-import org.apache.spark.sql.types.{DecimalType, LongType, StringType}
+import org.apache.spark.sql.types.{DataType, DecimalType, LongType, StringType}
 import org.apache.spark.sql.{Column, DataFrame}
 import za.co.absa.atum.agent.core.MeasurementProcessor
 import za.co.absa.atum.agent.core.MeasurementProcessor.{MeasurementFunction, ResultOfMeasurement}
 import za.co.absa.atum.model.dto.MeasureResultDTO.ResultValueType
-import za.co.absa.spark.commons.implicits.StructTypeImplicits.StructTypeEnhancements
 
 /**
  *  Type of different measures to be applied to the columns.
  */
-sealed trait Measure extends MeasurementProcessor with MeasureType {
-  val controlCol: String
+sealed trait Measure {
+  val measureName: String
+  def controlColumns: Seq[String]
```

Comment:
I would even dare to say, let's rename it to `measuredColumns`. @benedeki WDYT?

Reply:
Ok we agreed that it's what we want, but not now and here, but in https://github.com/AbsaOSS/atum-service/issues/116